src/FakeItEasy/IArgumentConstraintManager.cs

```diff
@@ -7,21 +7,28 @@ namespace FakeItEasy
     /// Manages attaching of argument constraints.
     /// </summary>
     /// <typeparam name="T">The type of argument to constrain.</typeparam>
-    [SuppressMessage("StyleCop.CSharp.DocumentationRules", "SA1623:PropertySummaryDocumentationMustMatchAccessors", Justification = "Fluent API.")]
     public interface IArgumentConstraintManager<T> : IHideObjectMembers
     {
         /// <summary>
-        /// Inverts the logic of the matches method.
-        /// </summary>
-        [SuppressMessage("Microsoft.Naming", "CA1716:IdentifiersShouldNotMatchKeywords", MessageId = "Not", Justification = "Part of the fluent syntax.")]
-        IArgumentConstraintManager<T> Not { get; }
-
-        /// <summary>
         /// Constrains the argument with a predicate.
         /// </summary>
         /// <param name="predicate">The predicate that should constrain the argument.</param>
         /// <param name="descriptionWriter">An action that will be write a description of the constraint.</param>
         /// <returns>A dummy argument value.</returns>
         T Matches(Func<T, bool> predicate, Action<IOutputWriter> descriptionWriter);
     }
+
+    /// <summary>
+    /// Enables negation of argument constraints.
+    /// </summary>
+    /// <typeparam name="T">The type of argument to constrain.</typeparam>
+    public interface INegatableArgumentConstraintManager<T> : IArgumentConstraintManager<T>
+    {
+        /// <summary>
+        /// Inverts the logic of the matches method.
+        /// </summary>
+        [SuppressMessage("Microsoft.Naming", "CA1716:IdentifiersShouldNotMatchKeywords", MessageId = "Not", Justification = "Part of the fluent syntax.")]
+        [SuppressMessage("StyleCop.CSharp.DocumentationRules", "SA1623:PropertySummaryDocumentationMustMatchAccessors", Justification = "Fluent API.")]
+        IArgumentConstraintManager<T> Not { get; }

```

Comment:
A better option would probably be to change the extension methods to extend `INegatableArgumentConstraintManager<T>` instead of `IArgumentConstraintManager<T>`